File: fishControl/src/main/java/br/com/getservicos/fishControl/service/api/TabelaCultivoService.java
package br.com.getservicos.fishControl.service.api;

import br.com.getservicos.fishControl.core.service.api.BaseCrudService;
import br.com.getservicos.fishControl.model.TabelaCultivo;
import br.com.getservicos.fishControl.model.Tanque;

public interface TabelaCultivoService extends BaseCrudService<TabelaCultivo, Integer> {
    TabelaCultivo getByTanque(Tanque tanque);
}

File: fishControl-Angular/src/app/services/tabela.cultivo.service.ts
import {BaseCrudService} from "../core/base.crud.service";
import {TabelaCultivo} from "../model/tabela.cultivo";
import {Injectable} from "@angular/core";
import {HttpClient} from "@angular/common/http";
import {Tanque} from "../model/tanque";
import {Observable} from "rxjs";

@Injectable({
  providedIn: "root"
})
export class TabelaCultivoService extends BaseCrudService<TabelaCultivo> {

  constructor(public http: HttpClient) {
    super(http, "/tabela-cultivo");
  }

  getByTanque(id: number) {
    return this._http.get<TabelaCultivo>(this.API + this._path + "/tanque/" + id);
  }
}
